### PR TITLE
k8s guard UX + onboarding-friction fixes (mentor-mode UX tweaks)

### PR DESCRIPTION
**Pull request by agent-refr**
> **AI-assisted change proposal.** Filed by agent driven by @Cervator via [GDD](https://siliconsaga.github.io/yggdrasil/gdd/).

## Summary

UX + correctness fixes for the mentoring/k8s feature (PR #111) and a few onboarding-friction items, all found dogfooding GDD as a new user on a fresh Windows machine. Every change is test-first (RED→GREEN); full workspace suite green (524/524).

**k8s guard:**
- `-f` manifests with a native Windows path (`C:\…`) no longer fail closed with a misleading "not found" — paths are normalized before the on-disk/yq check.
- Guard rejections are now classed (`scope` / `unbounded` / `precondition` / `context`) and rendered by one shared helper used by both the wrapper and the hook, so remediation advice fits the cause (widening the scope is no longer suggested for a cluster-scoped write or a malformed manifest) and the wrapper/hook wording can't drift.
- In-scope namespace lifecycle: `create`/`delete namespace` is allowed when the namespace name is in scope; `scope set` now arms on not-yet-existing namespaces (warns instead of failing), enabling the arm-then-create workflow across environments.
- Agent path: the hook now auto-allows raw in-scope `kubectl` reads (was: blanket redirect), denies out-of-scope writes with the full class-aware message, and still redirects in-scope writes to `ws k8s` for `--context` injection.
- `ws k8s --help` shows guard/scope help instead of forwarding to kubectl; `k8s` now appears in the `ws orient` survey (dispatch-parser regex excluded digit-named handlers).

**Tooling / onboarding:**
- New `ws gh` / `ws glab` wrappers inject the `.env` token so agent sessions don't fall through to an interactive `gh/glab auth login`; the token is never printed.
- `ws orient` and `ws audit-permissions` (both MUST-run at session start) are now allowlisted, so a fresh session doesn't prompt on them.
- `ws session get` prints a trailing newline; a loud comment in `ws-test.sh` flags that it sources `.env` (which can leak tokens into env-sensitive tests).
- Docs: added a "Kubernetes practice guard" section to the features catalog (it was only in skill/agent-training docs); updated `permissions.md` and the `gdd-k8s` skill.

## Test plan

- [ ] CI green (CodeRabbit + Copilot review requested).
- [ ] `ws test yggdrasil` passes locally (524/524 at branch tip).
- [ ] Manual: re-run TestPrimer-mentoring-k8s.md §1–§7 against a real cluster, incl. the §5 ambient-guard check from a plain terminal (the one item not auto-tested).

## Related

- Follows up PR #111 (mentoring/k8s training wheels); addresses the thelio-testing dogfooding findings.

## Summary by CodeRabbit

* **New Features**
  * Added `ws gh` and `ws glab` wrappers to run GitHub/GitLab CLI with workspace token support.
  * Enhanced Kubernetes `ws k8s` scope capture to handle pre-creating namespaces and arm contexts/namespace scopes with warnings for missing namespaces.
* **Bug Fixes**
  * Improved Kubernetes guard behavior for raw `kubectl` and cluster-scoped actions; tightened manifest `-f` handling (including Windows paths).
  * Made rejection/remediation messaging more consistent via shared rendering.
* **Documentation**
  * Updated Kubernetes guard practice and permissions documentation, including new command allow/deny guidance.
* **Tests**
  * Added/expanded coverage for k8s scoping/guard edge cases, wrapper passthrough/help behavior, and session output formatting.

### CODE REVIEW

File: .claude/hooks/gdd-permission-hook.sh [hunk 1]
```diff
@@ -941,11 +941,27 @@ for _entry in ${scoped_redirect_commands[@]+"${scoped_redirect_commands[@]}"}; d
...
+    # (a2) raw kubectl → route by guard verdict so an in-scope READ auto-allows
+    # (reads are free cluster-wide; forcing a redirect for a harmless read is
+    # pure friction). A blocked command denies with the guard's own class-aware
+    # message instead of the bare redirect text. An in-scope WRITE deliberately
+    # falls through to the (b) redirect so it runs via `ws k8s`, which injects
+    # --context — a raw write hitting the current-but-wrong context is exactly
+    # the accident the guard exists to prevent.
+    if [[ "$match_cmd" == kubectl\ * || "$match_cmd" == kubectl ]]; then
+        # shellcheck disable=SC2086
+        _sr_kverdict="$(k8s_guard_evaluate "$_sr_ctx" "$_sr_ns" $match_cmd 2>/dev/null || true)"
```
[the remark this answers is not in the document]
Comment (agent-refr) [reply .claude/hooks/gdd-permission-hook.sh:958]: Intentionally not changing this. The hook only receives the command as a string (.tool_input.command) — there is no tokenized argv to forward. Faithfully re-tokenizing a quoted string would require eval on agent-controlled input, which is unsafe and defeats the hook's purpose. The unquoted split only degrades a quoted arg containing spaces (realistically a -f path), and it fails CLOSED: a mangled path resolves to file-not-found, i.e. BLOCK:precondition. It cannot flip to fail-open, because namespace and pod names cannot contain spaces, so a write that should block can never become an allow. The 'ws k8s' wrapper path receives real argv and handles spaces correctly. This only affects the raw-kubectl accident-prevention path, which is best-effort by design. The split is already marked '# shellcheck disable=SC2086' to flag it as intentional.

File: scripts/ws-k8s-guard.sh [hunk 2]
```diff
@@ -67,8 +92,14 @@ k8s_guard_evaluate() {
...
             -f?*) ffiles+=("${a#-f}");;      # attached short form: -f<file>
+            # Known value-taking flags: consume the FOLLOWING token as the flag's
+            # value so it isn't mistaken for a positional (e.g. a namespace name
+            # in a create/delete lifecycle op — `delete namespace foo --timeout 5s`
+            # must not read `5s` as a second namespace). Attached (`-oyaml`) and
+            # equals (`--timeout=5s`) forms are single tokens and fall to `-*)`.
+            -o|--output|--timeout|--grace-period|-l|--selector|--field-selector) i=$((i+2)); continue ;;
             -*) : ;;
-            *) if [[ -z "$verb" ]]; then verb="$a"; else [[ -z "$verb2" ]] && verb2="$a"; fi ;;
+            *) if [[ -z "$verb" ]]; then verb="$a"; elif [[ -z "$verb2" ]]; then verb2="$a"; else rest_pos+=("$a"); fi ;;
```
[the remark this answers is not in the document]
Comment (agent-refr) [reply scripts/ws-k8s-guard.sh:102]: Not changing this — the suggested fix would introduce a fail-open. kubectl's --dry-run flag has a NoOptDefVal, so the space form 'kubectl delete namespace alice-sandbox --dry-run client' does NOT consume 'client': kubectl itself parses 'client' as a second positional, i.e. a second namespace target (it would try to delete namespaces alice-sandbox AND client). So the guard treating 'client' as a namespace name and blocking it when out of scope is correct, and matches kubectl's own parsing. If --dry-run were added to the value-consuming list, the guard would swallow 'client' and ALLOW the out-of-scope deletion — a fail-open. The correct mode form, --dry-run=client, is a single token and is already handled. Added regression tests in tests/ws-k8s/guard.bats (commit 66cfb61) asserting the space form blocks the out-of-scope name and the equals form is unaffected.

**Review by Cervator — APPROVED** [review]
